docroot/modules/custom/dcc_gtd_registration/src/Form/GlobalTrainingRegistrationForm.php

```diff
@@ -0,0 +1,482 @@
+   *  The form id.
+   */
+  public function getFormId() {
+    return 'dcc_gtd_registration';
+  }
+
+  /**
+   * Creates the form elements regarding the step.
+   *
+   * @param array $form
+   *  The form elements.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *  The form state.
+   *
+   * @return array
+   *  Return the $form with all the fields created according step.
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $step = $form_state->get("step");
+    if (!isset($step)) {
+      $form_state->set('step', 1);
+      $step = $form_state->get("step");
+    }
+    //If next button is pushed, save the values and increments the step.
+    if ($form_state->getTriggeringElement()['#value'] == 'Next') {
+      if ($step == 1) {
+        $form_state->set("first_name", $form_state->getValue("first_name"));
+        $form_state->set("last_name", $form_state->getValue("last_name"));
+        $form_state->set("email", $form_state->getValue("email"));
+        $form_state->set("drupal_user", $form_state->getValue("drupal_user"));
+        $form_state->set("phone", $form_state->getValue("phone"));
+        $form_state->set("age", $form_state->getValue("age"));
+        $form_state->set("gender", $form_state->getValue("gender"));
+      }
+      if ($step == 2) {
+        $form_state->set("country", $form_state->getValue("country"));
+        $form_state->set("city", $form_state->getValue("city"));
+        $form_state->set("address", $form_state->getValue("address"));
+      }
+      if ($step == 3) {
+        $form_state->set("occupation", $form_state->getValue("occupation"));
+        $form_state->set("organization", $form_state->getValue("organization"));
+        $form_state->set("industry_experience", $form_state->getValue("industry_experience"));
+      }
+
+      $form_state->set('step', $form_state->get('step') + 1);
+    }
+    // If back button is pushed, decrements the step.
+    if ($form_state->getTriggeringElement()['#value'] == 'Back') {
+      $form_state->set('step', $form_state->get('step') - 1);
+    }
+
+    $form['container'] = array(
+      '#type' => 'container',
+      '#attributes' => array(
+        'id' => 'ajaxcontainer',
+      ),
+    );
+
+    $step = $form_state->get('step');
+    switch ($step) {
+      case 1:
+        $this->stepOneElements($form,$form_state);
+        break;
+
+      case 2:
+        $this->stepTwoElements($form,$form_state);
+        break;
+
+      case 3:
+        $this->stepThreeElements($form,$form_state);
+        break;
+
+      case 4:
+        $this->stepFourElements($form,$form_state);
+        break;
+
+      default:
+        break;
+
+    }
+    if ($step == 1 || $step == 2 || $step == 3) {
+      $form['container']['next'] = array(
+        '#type' => 'button',
+        '#value' => 'Next',
+        '#ajax' => array(
+          'callback' => 'Drupal\dcc_gtd_registration\Form\GlobalTrainingRegistrationForm::ajax',
```

Comment:
Here the callback value I  think can have a shorter version [$this, 'ajax']. Please test it.